server/task_check_scheduler.go

```diff
@@ -177,17 +177,13 @@ func (s *TaskCheckScheduler) Register(taskType string, executor TaskCheckExecuto
 	s.executors[taskType] = executor
 }
 
-// shouldScheduleTimingTaskCheck will return whether should we schedule a timing task check for current task
-// the logic goes like the following:
-// 1. there is no running timing task check
-// 2(a). the earliestAllowedTs state has been altered since last check (either via new creation or patch).
-// 2(b). the earliestAllowedTs has NOT been altered since last check, however the last check had failed && earliestAllowedTs has passed now.
-// ONLY when {1 && (2.a || 2.b)} listed above or EXPLICITLY set the forceSchedule to TRUE would this function return true
+// Why this comment is such long and tedious? It a long story...
+// TL;DR
+// 1. we do not want anything happen if user does not set this field at the frontend.
+// 2. If user has specified this field once (even if she set it to default latter), we will schedule one ever since.
+// 3. We only want to schedule one if the payload field is different to the last run (unless user force one)
+// 4. once the specified time has passed, schedule one immediately, so the task will no longger be blocked
```

Comment:
Capitalize the first letter once -> Once

>> once the specified time has passed, schedule one immediately, so the task will no longger be blocked

Once the specified time has passed, schedule one immediately to unblock the task.